### ISSUE TITLE
no-redeclare: false positives for type definitions

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/typescript-eslint-parser#443]
**What version of TypeScript are you using?**
2.6.1

**What version of `typescript-eslint-parser` are you using?**
12.0.0

**What code were you trying to parse?**

```ts
const Foo = 1;
// unexected error: 'Foo' is already defined. (no-redeclare)
type Foo = 1;
```

We've decided to disable this ESLint rule as TypeScript also provides similar functionality anyway, but logging this just for others/reference.

For context, this pattern is common when using [unionize](https://github.com/pelotom/unionize):

``` ts
import { unionize, ofType } from 'unionize'

export const Action = unionize({
  ADD_TODO:                ofType<{ id: string; text: string }>(),
  SET_VISIBILITY_FILTER:   ofType<'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED'>(),
  TOGGLE_TODO:             ofType<{ id: string }>(),
});

export type Action = typeof Action._Union;
```

**Comment by OliverJAsh** [comment]
For context, this pattern is common when using [unionize](https://github.com/pelotom/unionize):

``` ts
import { unionize, ofType } from 'unionize'

// Define a record mapping tag literals to value types
const Action = unionize({
  ADD_TODO:                ofType<{ id: string; text: string }>(),
  SET_VISIBILITY_FILTER:   ofType<'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED'>(),
  TOGGLE_TODO:             ofType<{ id: string }>(),
}, 'type', 'payload'); // optionally override tag and value property names

type Action = typeof Action._Union;
```
